pkg/controller/cassandradatacenter/controller/reconciler.go

```diff
@@ -0,0 +1,107 @@
+package controller
+
+import (
+	"context"
+	cassandraoperatorv1alpha1 "github.com/instaclustr/cassandra-operator/pkg/apis/cassandraoperator/v1alpha1"
+	"k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("CassandraDataCenterReconciler")
+
+// CassandraDataCenterReconciler reconciles a CassandraDataCenter object
+type CassandraDataCenterReconciler struct {
+	// This client is a split client that reads objects from the cache and writes to the apiserver
+	client client.Client
```

Comment:
Can we keep a pointer here too?

Reply:
maybe, I am not sure if that was not generated by operator-sdk, I have to check

Reply:
i just leave it as it is for now